go/cmd/dolt/commands/reset.go

```diff
@@ -80,90 +83,109 @@ func (cmd ResetCmd) ArgParser() *argparser.ArgParser {
 	return cli.CreateResetArgParser()
 }
 
+func (cmd ResetCmd) RequiresRepo() bool {
+	return false
+}
+
 // Exec executes the command
 func (cmd ResetCmd) Exec(ctx context.Context, commandStr string, args []string, dEnv *env.DoltEnv, cliCtx cli.CliContext) int {
 	ap := cli.CreateResetArgParser()
 	help, usage := cli.HelpAndUsagePrinters(cli.CommandDocsForCommandString(commandStr, resetDocContent, ap))
 	apr := cli.ParseArgsOrDie(ap, args, help)
 
-	if dEnv.IsLocked() {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(env.ErrActiveServerLock.New(dEnv.LockFile())), help)
-	}
-
-	roots, err := dEnv.Roots(ctx)
+	queryist, sqlCtx, closeFunc, err := cliCtx.QueryEngine(ctx)
 	if err != nil {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+		cli.Println(err.Error())
+		return 1
+	}
+	if closeFunc != nil {
+		defer closeFunc()
 	}
 
 	if apr.ContainsAll(HardResetParam, SoftResetParam) {
 		verr := errhand.BuildDError("error: --%s and --%s are mutually exclusive options.", HardResetParam, SoftResetParam).Build()
 		return HandleVErrAndExitCode(verr, usage)
 	} else if apr.Contains(HardResetParam) {
-		return handleResetHard(ctx, apr, usage, dEnv, roots)
-	} else {
-		if apr.NArg() == 1 {
-			ref := apr.Arg(0)
-			isValidRef, err := actions.IsValidRef(ctx, ref, dEnv.DoltDB, dEnv.RepoStateReader())
-			if err != nil {
-				return handleErrAndExit(err)
-			}
-			if isValidRef {
-				return handleResetSoftToRef(ctx, dEnv, ref, usage)
-			}
+		if apr.NArg() > 1 {
+			return handleResetError(fmt.Errorf("--hard supports at most one additional param"), usage)
 		}
-
-		return handleResetSoftTables(ctx, apr.Args, roots, dEnv, usage)
 	}
-}
 
-func handleResetSoftTables(ctx context.Context, tables []string, roots doltdb.Roots, dEnv *env.DoltEnv, usage cli.UsagePrinter) int {
-	roots, err := actions.ResetSoft(ctx, dEnv.DbData(), tables, roots)
+	// process query through prepared statement to prevent sql injection
+	query, params := constructParametrizedDoltResetQuery(apr)
+	interpolatedQuery, err := dbr.InterpolateForDialect(query, params, dialect.MySQL)
```

Comment:
I suggest interpolating the string in the construct method. Easier than getting to values back only to pass them into another function on the next line.